cmd/soroban-cli/src/commands/contract/info/shared.rs

```diff
@@ -0,0 +1,82 @@
+use std::path::PathBuf;
+
+use clap::arg;
+use soroban_env_host::xdr;
+use soroban_rpc::Client;
+
+use crate::commands::contract::info::shared::Error::InvalidWasmHash;
+use crate::commands::contract::InfoOutput;
+use crate::config::{locator, network};
+use crate::utils::rpc::get_remote_wasm_from_hash;
+use crate::wasm;
+use crate::wasm::Error::ContractIsStellarAsset;
+
+#[derive(Debug, clap::Args, Clone, Default)]
+#[command(group(
+    clap::ArgGroup::new("src")
+    .required(true)
+    .args(& ["wasm", "wasm_hash", "contract_id"]),
+))]
+#[group(skip)]
+pub struct Args {
+    /// Wasm file to extract the data from
+    #[arg(long, group = "src")]
+    pub wasm: Option<PathBuf>,
+    /// Wasm hash to get the data for
+    #[arg(long = "wasm-hash", group = "src")]
+    pub wasm_hash: Option<String>,
+    /// Contract id to get the data for
+    #[arg(long = "id", env = "STELLAR_CONTRACT_ID", group = "src")]
```

Comment:
I'm inclined to leave off the env for this command. Envs are useful for setting for use across many commands, where-as this command is quite specific.
```suggestion
    #[arg(long = "id", group = "src")]
```

Reply:
Oh right, we already use this env consistently across the cli. Please ignore the comment above.